compose/service.py

```diff
@@ -430,12 +430,16 @@ def start_container(self, container):
         return container
 
     def connect_container_to_networks(self, container):
-        one_off = (container.labels.get(LABEL_ONE_OFF) == "True")
+        connected_networks = container.get('NetworkSettings.Networks')
 
         for network in self.networks:
+            if network in connected_networks:
+                self.client.disconnect_container_from_network(
+                    container.id, network)
```

Comment:
Good to know, thanks!

Might be worth making a feature request on engine for a way to accomplish this without connecting to a network originally.